@here/harp-datasource-protocol/lib/StringEncodedNumeral.ts

```diff
@@ -64,14 +64,12 @@ export const StringEncodedRGB: StringEncodedNumeralFormat = {
     regExp: /rgb\((?:([0-9]{1,2}|1[0-9]{1,2}|2[0-4][0-9]|25[0-5]), ?)(?:([0-9]{1,2}|1[0-9]{1,2}|2[0-4][0-9]|25[0-5]), ?)(?:([0-9]{1,2}|1[0-9]{1,2}|2[0-4][0-9]|25[0-5]))\)/,
     decoder: (encodedValue: string) => {
         const channels = StringEncodedRGB.regExp.exec(encodedValue)!;
-        tmpColor
-            .setRGB(
-                parseInt(channels[1], 10) / 255,
-                parseInt(channels[2], 10) / 255,
-                parseInt(channels[3], 10) / 255
-            )
-            .getHSL(tmpHSL);
-        return [tmpHSL.h, tmpHSL.s, tmpHSL.l];
+        tmpColor.setRGB(
```

Comment:
are all these regex tests executed twice? can we remove that ? these are quite expensive tests and we will run several 100s of them for each render call.

Reply:
Could we agree to do this in separate PR/ticket? I have already another PR that is dependent on that and making that optimization here will conflict related stuff a lot.

Reply:
BTW, there is another slight optimization that I would like to do, mentioned in the comment above... Pass the output array to decoder as parameter instead of creating temp objects all the time.

Reply:
sure.